app/views/govuk_publishing_components/components/_cards.html.erb

```diff
@@ -1,57 +1,56 @@
 <%
+  shared_helper = GovukPublishingComponents::Presenters::SharedHelper.new(local_assigns)
   heading ||= nil
   items ||= nil
   sub_heading_level ||= 3
   two_column_layout ||= false
   local_assigns ||= nil
+  data_attributes ||= nil
 
   ul_classes = %w[gem-c-cards__list]
   ul_classes << 'gem-c-cards__list--two-column-desktop' if two_column_layout
 
-  shared_helper = GovukPublishingComponents::Presenters::SharedHelper.new(local_assigns)
+  def is_tracking?(items)
+    if items.any?
+      items.each do |item|
+        return true if item[:link][:data_attributes]
+      end
+      false
+    end
+  end
 %>
+
 <% if items.present? %>
   <div class="gem-c-cards">
     <% if heading %>
       <%= content_tag(shared_helper.get_heading_level, class: "gem-c-cards__heading govuk-heading-m") do %>
         <%= heading %>
       <% end %>
-    <% end %>
-    <%= content_tag 'ul', class: ul_classes do %>
+    <% end %>    
+
+    <%= content_tag('ul', {
+    class: ul_classes,
+      data: {
+        module: ("gem-track-click" if is_tracking?(items))
```

Comment:
I think this approach is over-complicating things. Normally we just pass attributes to a component, then wrap that component instance in the `gem-track-click` module, i.e.

```
<div data-module="gem-track-click">
  <%= render "govuk_publishing_components/components/cards", {
    heading: "Topics",
    items: items,
  } %>
</div>
```

This approach also allows us to separate out the testing - `gem-track-click` can have its own tests to ensure its working, and the component only needs to be tested for the ability to receive and display data attributes.

Reply:
Ohh, I was partially [basing it on this](https://github.com/alphagov/govuk_publishing_components/blob/main/lib/govuk_publishing_components/presenters/image_card_helper.rb#L28) I thought being dynamic was desirable only being added when there's attributes.

Ok I'll strip on this `gem-track-click` stuff and keep the other changes.